pkg/apis/core/validation/validation_test.go

```diff
@@ -1926,6 +1942,42 @@ func testValidatePVC(t *testing.T, ephemeral bool) {
 				},
 			}),
 		},
+		"invaild-apigroup-in-data-source": {
+			isExpectedFailure: true,
```

Comment:
I know this is how the current tests are written, but I think it would be good to validate that the error is actually the correct type and for this field

Reply:
@msau42 Should I do it in this pr or a follow-up pr?

Reply:
I think a followup PR could work because it would touch many more tests